Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp

```diff
@@ -101,84 +101,74 @@ Ref<MediaPromise> RemoteSourceBufferProxy::sourceBufferPrivateDidReceiveInitiali
 
     segmentInfo.audioTracks = segment.audioTracks.map([&](auto& audioTrackInfo) {
         auto id = audioTrackInfo.track->id();
-        m_remoteMediaPlayerProxy->addRemoteAudioTrackProxy(*audioTrackInfo.track);
+        m_remoteMediaPlayerProxy->addRemoteAudioTrackProxy(Ref { *audioTrackInfo.track });
         m_mediaDescriptions.try_emplace(id, *audioTrackInfo.description);
         return InitializationSegmentInfo::TrackInformation { MediaDescriptionInfo(*audioTrackInfo.description), id };
     });
 
     segmentInfo.videoTracks = segment.videoTracks.map([&](auto& videoTrackInfo) {
         auto id = videoTrackInfo.track->id();
-        m_remoteMediaPlayerProxy->addRemoteVideoTrackProxy(*videoTrackInfo.track);
+        m_remoteMediaPlayerProxy->addRemoteVideoTrackProxy(Ref { *videoTrackInfo.track });
         m_mediaDescriptions.try_emplace(id, *videoTrackInfo.description);
         return InitializationSegmentInfo::TrackInformation { MediaDescriptionInfo(*videoTrackInfo.description), id };
     });
 
     segmentInfo.textTracks = segment.textTracks.map([&](auto& textTrackInfo) {
         auto id = textTrackInfo.track->id();
-        m_remoteMediaPlayerProxy->addRemoteTextTrackProxy(*textTrackInfo.track);
+        m_remoteMediaPlayerProxy->addRemoteTextTrackProxy(Ref { *textTrackInfo.track });
         m_mediaDescriptions.try_emplace(id, *textTrackInfo.description);
         return InitializationSegmentInfo::TrackInformation { MediaDescriptionInfo(*textTrackInfo.description), id };
     });
 
     // We need to wait for the CP's MediaPlayerRemote to have created all the tracks
-    return m_remoteMediaPlayerProxy->commitAllTransactions()->whenSettled(RunLoop::current(), [weakThis = ThreadSafeWeakPtr { *this }, this, segmentInfo = WTFMove(segmentInfo)](auto&& result) mutable -> Ref<MediaPromise> {
+    return m_remoteMediaPlayerProxy->commitAllTransactions()->whenSettled(Ref { RunLoop::current() }, [weakThis = ThreadSafeWeakPtr { *this }, this, segmentInfo = WTFMove(segmentInfo)](auto&& result) mutable -> Ref<MediaPromise> {
```

Comment:
We should probably rename `RunLoop::current` to `RunLoop::currentSingleton` to suppress the warning.

Reply:
Given Basuke's suggestion below, do we prefer using `RunLoop::protectedCurrent()` instead of renaming `current` to `currentSingleton` later on?

Reply:
We can adopt protectedCurrent yeah, but we should probably get rid of this function altogether and just rename the function to `currentSingleton`. It's pretty silly to ref counting a (thread local) singleton object.

Reply:
Got it--that makes sense. In that case, I'll revert the `RunLoop::current` changes I've made in this patch and write a patch after this that just renames the function to `currentSingleton` and use that everywhere.